### DISCUSSION TITLE
Having trouble compiling stb_image on clang in C. Spits out 'expected expression' errors

### DISCUSSION THREAD

**Discussion by AliElSaleh** [nothings/stb#1431] [Q&A] [answered]
It's really odd. I dont have a clue why it is reporting an error like this

![image](https://user-images.githubusercontent.com/19608222/210604010-2179566b-b617-4892-92cd-f6dcc5072ca0.png)

And no, it's not because of the SIMD stuff. I tried with and without STBI_NO_SIMD. I still get the same 'expected expression' errors

![image](https://user-images.githubusercontent.com/19608222/210605959-aee3fb5d-e121-4bf8-8e79-b9c1d23a24bd.png)

A pattern i'm noticing is that it seems to get stumped at the 'out' ptr

The C std im using is -std=c2x. This is because i get a warning about 'omitting a parameter name', which again, doesn't look like it is lol

![image](https://user-images.githubusercontent.com/19608222/210610291-fd4d5d31-fd27-48da-bcb7-9418e484c2ec.png)

I'm using the library like normal by defining the implementation macro followed by the include to stb_image.h

Any help is appreciated, thanks!

**Comment by rygorous** [comment, marked as answer]
It sounds like `out` got added as a keyword or preprocessor `#define`, but I have no idea what it maps to.

As a general matter of policy, we'll happily take patches to fix issues with officially published standard revisions, but C2X is still in progress and current implementations are subject to change. For the time being, I'd suggest to compile the translation unit that defines `STB_IMAGE_IMPLEMENTATION` with C17 or below.

**Reply by AliElSaleh** [reply]
ah ok, that explains it. i've removed the `out` macro from my code base. it compiles now. thanks for the help!

**Reply by AliElSaleh** [reply]
and yeah i dont want to use C2X, i had been using C11 until this issue arose. will switch to C17
